gremlin-javascript/src/main/javascript/gremlin-javascript/test/integration/socket-connection-tests.js

```diff
@@ -49,7 +49,49 @@ describe('Connection', function () {
   });
 
   describe('#open()', function () {
+    it('should use the ws WebSocket when ws specific options are provided', function () {
+      let globalWebsocketCalls = 0;
+      globalThis.WebSocket = function () {
+        globalWebsocketCalls++;
+      };
+      const wsSpecificOptions = [
+        'headers',
+        'ca',
+        'cert',
+        'pfx',
+        'rejectUnauthorized',
+        'agent',
+        'perMessageDeflate',
+      ];
+      const allOptionTests = wsSpecificOptions.map((wsOption) => {
+        const connection = helper.getDriverRemoteConnection(`ws://localhost:${testServerPort}/401`, {
+          [wsOption]: 'this option is set',
+        });
+        return connection.open();
+      });
+      Promise.allSettled(allOptionTests).then(function () {
```

Comment:
If I'm not mistaken, this promise should be returned from the test function so that Jest waits on it properly.

Reply:
Good catch, pushed the fix for this! Thank you :)